Document/0x06h-Testing-Platform-Interaction.md

```diff
@@ -3,365 +3,2678 @@
+
+There is also a *visual* way to inspect the status of some app permissions when using the iPhone/iPad by opening "Settings" and scrolling down until you find the app you're interested in. When clicking on it, this will open the "ALLOW APP_NAME TO ACCESS" screen. However, not all permissions might be displayed yet. You will have to *trigger* them in order to be listed on that screen.
+
+![Settings Allow App Screen](Images/Chapters/0x06h/settings_allow_screen.png)
+ 
+For example, if you're testing a messaging app, try to share something. As you do it, surely the app will offer options to share photos or your location. If you click, for example, on share location, this will trigger the permission dialogue and the permission will be visible in "Settings" as well.
+
+
+### Testing for Sensitive Functionality Exposure Through IPC
+
+During implementation of a mobile application, developers may apply traditional techniques for IPC (such as using shared files or network sockets). The IPC system functionality offered by mobile application platforms should be used because it is much more mature than traditional techniques. Using IPC mechanisms with no security in mind may cause the application to leak or expose sensitive data.
+
+In contrast to Android's rich Inter-Process Communication (IPC) capability, iOS offers some rather limited options for communication between apps. In fact, there's no way for apps to communicate directly. In this section we will present the different types of indirect communication offered by iOS and how to test them. Here's an overview:
+
+- [Custom URL Schemes](#Custom-URL-Schemes)
+- [Universal Links](#Universal-Links)
+- [UIActivity Sharing](#UIActivity-Sharing)
+- [App Extensions](#App-Extensions)
+- [UIPasteboard](#UIPasteboard)
+
+#### Custom URL Schemes
+
+Please refer to the next section ["Testing Custom URL Schemes"](#Testing-Custom-URL-Schemes) for more information on what custom URL schemes are and how to test them.
+
+#### Universal Links
+
+##### Overview
+
+Universal links the iOS equivalent to the Android App Links (aka. Digital Asset Links) and are used for deep linking. When a user taps a link to the app's website he will get seamlessly redirected to the corresponding installed app without going through Safari. If the app isn’t installed, the link will open in Safari.
+
+They were introduced as a way to *prevent* URL scheme hijacking attacks. According to the [Apple Developer Documentation](https://developer.apple.com/library/archive/documentation/General/Conceptual/AppSearch/UniversalLinks.html), universal links give several key benefits that are not applicable when using custom URL schemes and are the recommended way to implement deep linking. Specifically, universal links are:
```

Comment:
Add description for links (https://developer.apple.com/library/archive/documentation/General/Conceptual/AppSearch/UniversalLinks.html "Universal Links")

Reply:
done for the whole PR